mlflow/metrics/genai/genai_metric.py

```diff
@@ -558,4 +604,104 @@ def eval_fn(
         version=version,
         metric_details=evaluation_context["eval_prompt"].__str__(),
         metric_metadata=metric_metadata,
+        genai_metric_args=genai_metric_args,
     )
+
+
+def _filter_by_field(df, field_name, value):
+    return df[df[field_name] == value]
+
+
+def _deserialize_genai_metric_args(args_dict):
+    mlflow_version_at_ser = args_dict.get("mlflow_version")
+    fn_name = args_dict.get("fn_name")
+    if fn_name is None or mlflow_version_at_ser is None:
+        raise MlflowException(
+            message="The artifact JSON file appears to be corrupted and cannot be deserialized. "
+            "Please regenerate the custom metrics and rerun the evaluation. "
+            "Ensure that the file is correctly formatted and not tampered with.",
+            error_code=PUBLIC_UNDOCUMENTED,
+        )
+
+    if mlflow_version_at_ser != VERSION:
+        warnings.warn(
+            f"The custom metric definitions were serialized using MLflow {mlflow_version_at_ser}. "
+            f"Deserializing them with the current version {VERSION} might cause mismatches. "
+            "Please ensure compatibility or consider regenerating the metrics "
+            "using the current version.",
+            UserWarning,
+            stacklevel=2,
+        )
+
+    args_dict.pop("mlflow_version")
+    args_dict.pop("fn_name")
+    if fn_name == make_genai_metric_from_prompt.__name__:
+        return make_genai_metric_from_prompt(**args_dict)
+
+    examples = args_dict["examples"]
+    if examples is None:
+        return make_genai_metric(**args_dict)
+
+    deser_examples = []
+    for example in examples:
+        deser_examples.append(EvaluationExample(**example))
```

Comment:
Could we do a list comprehension here?

```suggestion
    deser_examples = [EvaluationExample(**example) for example in examples]
```

Reply:
TIL